File: internal/command/command.go
package command

import (
	"fmt"
)

type Command interface {
	Execute()
}

type command struct {
	name string
}

type CommandHandler interface {
	Handle(Command)
}

type commandHandler struct {
	name string
}

type CommandDispatcher struct {
	handlers map[string]CommandHandler
}

func New() Command {
	return command{name: "Create"}
}

func NewHandler() CommandHandler {
	return commandHandler{name: "Create"}
}

func (c command) Execute() {
	fmt.Println("Executing command: ", c.name)
}

func (c commandHandler) Handle(command Command) {
	command.Execute()
}

File: cqrs/src/command.rs

trait Command {}
trait CommandHandler {
    fn execute(&self, command: Box<dyn ICommand>);
}
struct CommandDispatcher {
    handlers: HashMap<String, Box<dyn ICommandHandler>>,
}

impl CommandDispatcher {
    fn dispatch(&self, command: Box<dyn ICommand>) {
        let command_type = format!("{:?}", command);
        if let Some(handler) = self.handlers.get(&command_type) {
            handler.execute(command);
        } else {
            println!("handler not found", command_type);
        }
    }


}